src/modules/package-index.ts

```diff
@@ -0,0 +1,135 @@
+
+export interface FunctionalInterfaceDescription {
+    name: string;
+    value: string;
+    args: Argument[];
+    default?: boolean;
+}
+
+export interface ExampleDescription {
+    name: string;
+    value: string;
+    language: string;
+}
+
+export interface Argument {
+    id: string;
+    description: string;
+    type: string;
+    default: string;
+    required: boolean;
+    [key: string]: any;
+}
+
+export interface ExposedInterface {
+    type: string;
+    description: string;
+    args: Argument[];
+    default?: boolean;
+    [key: string]: any;
+}
+
+export interface PkgIndexEntry {
+    type: string;
+    package: string;
+    exposedInterfaces: ExposedInterface[];
+    [key: string]: any;
+}
+
+export class PackageIndex {
+    private _packages: PkgIndexEntry[];
+
+    private constructor(packages: PkgIndexEntry[]) {
+        this._packages = packages;
+    }
+
+    static read(path: string = './package-index.json'): PackageIndex {
+        try {
+            const packageIndexFile = readFileSync(path, DEFAULT_BUFFER_ENCODING);
+            const packageIndex: PkgIndexEntry[] = JSON.parse(packageIndexFile);
+            return new PackageIndex(packageIndex);
+        } catch (error) {
+            throw new Error('No package-index.json found.');
+        }
+    }
+
+    getCores(): PkgIndexEntry[] {
+        return this._packages.filter((pkg: PkgIndexEntry) => pkg.type === 'core');
+    }
+
+    getExtensions(): PkgIndexEntry[] {
+        return this._packages.filter((pkg: PkgIndexEntry) => pkg.type === 'extension');
+    }
+
+    getAvailableLanguages(): string[] {
+        const AVAILABLE_LANGUAGES: string[] = [];
+        this.getCores().forEach((packageEntry: PkgIndexEntry) => {
+            const match = packageEntry.package.match(sdkPattern);
+            if (match) {
+                const language = match[1];
+                AVAILABLE_LANGUAGES.push(language);
+            }
+        });
+        return AVAILABLE_LANGUAGES;
+    }
+
+    getAvailableExamples(): ExampleDescription[] {
+        const AVAILABLE_EXAMPLES: ExampleDescription[] = [];
+        this.getCores().forEach((packageEntry: PkgIndexEntry) => {
+            const match = packageEntry.package.match(sdkPattern);
+            if (match) {
+                const examples = packageEntry.exposedInterfaces.filter((exposed: ExposedInterface) => exposed.type === 'examples');
+                for (const example in examples[0].args) {
+                    const language = match[1];
+                    AVAILABLE_EXAMPLES.push({
+                        name: examples[0].args[example].description,
+                        value: examples[0].args[example].id,
+                        language: language,
+                    });
+                }
+            }
+        });
+        return AVAILABLE_EXAMPLES;
+    }
+
+    getAvailableInterfaces(): FunctionalInterfaceDescription[] {
+        const AVAILABLE_INTERFACES: FunctionalInterfaceDescription[] = [];
```

Comment:
Why constant naming style when contents are modified?

```suggestion
        const availableInterfaces: FunctionalInterfaceDescription[] = [];
```